### ISSUE TITLE
Support forthcoming UTF8 support in Prometheus

### ISSUE THREAD

**Issue by ywwg** [grafana/agent#5186]
### Request

We plan to implement UTF8 support in Prometheus metric and label names, as briefly outlined here: https://github.com/prometheus/prometheus/issues/12630

Grafana Agent will need updates to support the new exposition syntax and validation parameters, with a feature flag to enable/disable the behavior. Design doc forthcoming will be linked in this issue.

### Use case

Support for rich metric names makes Prometheus more future-proof, for example with Open Telemetry. More justification to be described in the Prometheus design proposal.

**Comment by tpaschalis** [comment]
Thanks for the heads-up, we're looking forward to using this change!

**Comment by rfratto** [comment]
> with a feature flag to enable/disable the behavior

Is a feature _flag_ strictly necessary @ywwg? Given how much behavior is available in the agent, we typically don't use feature flags for behavior specific to signals. 

(I personally wouldn't mind just enabling it downstream without a way to disable as a way of dogfooding the new functionality)

**Comment by ywwg** [comment]
No, a flag is not necessary.  As the design comes into focus we can figure out the best way to opt in/out.  (For example, we probably want to support sending utf8 metrics to an old version of prometheus that does not support it (via an escaping mechanism), and do that without needing to discover the compatibility via content negotiation)

**Comment by ywwg** [comment]
Proposal has been merged: https://github.com/prometheus/proposals/blob/main/proposals/2023-08-21-utf8.md
